src/Altinn.Platform/Altinn.Platform.Storage/Storage/Helpers/AuthorizationHelper.cs

```diff
@@ -307,9 +307,12 @@ public bool ContainsRequiredScope(string requiredScope, ClaimsPrincipal user)
 
             contextScope ??= user.Claims.Where(c => c.Type.Equals("scope")).Select(c => c.Value).FirstOrDefault();
 
-            if (!string.IsNullOrWhiteSpace(contextScope) && contextScope.Contains(requiredScope, StringComparison.InvariantCultureIgnoreCase))
+            foreach (string scope in requiredScope)
             {
-                return true;
+                if (!string.IsNullOrWhiteSpace(contextScope) && contextScope.Contains(scope, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
null check could be moved outside of the foreach loop I guess,

Reply:
done